src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java

```diff
@@ -129,17 +129,13 @@ public void visitToken(DetailAST detailAST) {
         if (currentNode.getType() == TokenTypes.BOR) {
             exceptionTypes.addAll(getAllExceptionTypes(currentNode));
             currentNode = currentNode.getNextSibling();
-            if (currentNode != null) {
-                exceptionTypes.add(currentNode);
-            }
+            exceptionTypes.add(currentNode);
         }
         else {
-            exceptionTypes.add(currentNode);
-            currentNode = currentNode.getNextSibling();
-            while (currentNode != null) {
+            do {
                 exceptionTypes.add(currentNode);
                 currentNode = currentNode.getNextSibling();
-            }
+            } while (currentNode != null);
```

Comment:
This change is just to simplify the code, and this is why I am going to do regression.